GP11 : Time functions fix/panic reason

cedric-chaumont-st-dev opened the pull request:
Signed-off-by: Cedric Chaumont <cedric.chaumont@st.com>

pascal-brand-st-dev commented:
Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

jenswi-linaro commented on core/tee/tee_svc.c:
```diff
@@ -897,12 +897,15 @@ TEE_Result tee_svc_get_time(enum utee_time_category cat, TEE_Time *mytime)
+		t.millis = 0;
 	}
 
+	res2 = tee_svc_copy_to_user(s, mytime, &t, sizeof(t));
```
Now we're always copying back the time even if we didn't get anything useful, why?

cedric-chaumont-st-dev replied:
If an error other than TEE_ERROR_OVERFLOW is returned, this structure is filled with zeroes.

jenswi-linaro replied:
How is that useful?

cedric-chaumont-st-dev commented:
Updated.

cedric-chaumont-st-dev commented:
Updated.

jenswi-linaro commented on lib/libutee/tee_api.c:
```diff
@@ -202,12 +202,37 @@ TEE_Result TEE_Wait(uint32_t timeout)
+	res = utee_get_time(UTEE_TIME_CAT_TA_PERSISTENT, time);
+
+	if (res != TEE_SUCCESS && res != TEE_ERROR_OVERFLOW) {
+		time->seconds = 0;
```
So now this has moved into this function instead. I still don't understand why we should set it to zero though. The application shouldn't have any expectations on what's in *time if the function doesn't return `TEE_SUCCESS` or `TEE_ERROR_OVERFLOW`.

cedric-chaumont-st-dev replied:
@jenswi-linaro: "If an error other than TEE_ERROR_OVERFLOW is returned, this structure is filled with zeroes." What is your understanding about this ?

jenswi-linaro replied:
Sorry, I missed that. Yes this is the way to do it.

jenswi-linaro commented:
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

cedric-chaumont-st-dev commented:
Rebased.Squashed.Tested (ATM, ARM Juno).
28701 subtests of which 0 failed
687 test cases of which 0 failed
0 test case was skipped
TEE test application done!
